src/_pytest/config/__init__.py

```diff
@@ -1143,16 +1138,22 @@ def _getini(self, name: str) -> Any:
                 if type is None:
                     return ""
                 return []
+        # coerce the values based on types
```

Comment:
OK, I see now that `_get_ini_config_from_pyproject_toml` converts the scalers to `str`. I suggest noting that here, otherwise it's non obvious, as I demonstrated :)